packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+                break;
+
+            case ViewportPositionerHorizontalPositionLabel.right:
+                translate = this.viewportRect.left - this.anchorRight;
+                translate = translate > 0 ? translate + 1 : 0;
+                break;
+        }
+        return translate;
+    };
+
+    /**
+     *  Calculate vertical tranlation to keep positioner in view
+     */
+    private calculateVerticalTranslate = (
+        verticalPosition: ViewportPositionerVerticalPositionLabel
+    ): number => {
+        if (!this.props.verticalAlwaysInView) {
+            return 0;
+        }
+
+        let translate: number = 0;
+
+        switch (verticalPosition) {
+            case ViewportPositionerVerticalPositionLabel.top:
+                translate = this.viewportRect.bottom - this.anchorTop;
+                translate = translate < 0 ? translate - 1 : 0;
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetTop:
+                translate = this.viewportRect.bottom - this.anchorBottom;
+                translate = translate < 0 ? translate - 1 : 0;
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.insetBottom:
+                translate = this.viewportRect.top - this.anchorTop;
+                translate = translate < 0 ? 0 : translate + 1;
+                break;
+
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                translate = this.viewportRect.top - this.anchorBottom;
+                translate = translate < 0 ? 0 : translate + 1;
+                break;
+        }
+
+        return translate;
+    };
+
+    /**
+     * Request's an animation frame if there are currently no open animation frame requests
+     */
+    private requestFrame = (): void => {
+        if (this.openRequestAnimationFrame === null && !this.state.disabled) {
+            this.openRequestAnimationFrame = window.requestAnimationFrame(
+                this.updateLayout
+            );
+        }
+    };
+
+    /**
+     * get the anchor element
+     */
+    private getAnchorElement = (): HTMLElement => {
+        if (isNil(this.props.anchor)) {
+            return null;
+        }
+
+        if (this.props.anchor instanceof HTMLElement) {
+            return this.props.anchor;
+        } else {
+            return this.props.anchor.current;
+        }
+    };
+
+    /**
+     * get the viewport element
+     */
+    private getViewportElement = (): HTMLElement => {
+        if (isNil(this.props.viewport)) {
+            return document.scrollingElement as HTMLElement;
+        }
+
+        if (this.props.viewport instanceof HTMLElement) {
+            return this.props.viewport;
+        } else {
+            return this.props.viewport.current;
+        }
+    };
+
+    /**
+     * Converts simple horizontal position to a position label based on AxisPositioningMode
+     */
+    private ConvertHorizontalPositionToLabel = (
```

Comment:
the first character should be lowercase, also since this is returning something I would prefer the addition of "get" so "getHorizontalPositionToLabel"

Reply:
yep